### ISSUE TITLE
bug: Table.select(col_name=True/False) gives unexpected behavior

### ISSUE THREAD

**Issue by NickCrews** [ibis-project/ibis#8255]
### What happened?

```python
import ibis

ibis.options.interactive = True

t = ibis.memtable(
    {
        "x": [1, 2, 3],
        "y": [4, 5, 6],
    }
)
t.select(false=False, true=True)
# ┏━━━━━━━┳━━━━━━━┓
# ┃ false ┃ true  ┃
# ┡━━━━━━━╇━━━━━━━┩
# │ int64 │ int64 │
# ├───────┼───────┤
# │     1 │     4 │
# │     2 │     5 │
# │     3 │     6 │
# └───────┴───────┘
```

I would expect the same result as if I'd used ibis.literal(). I thought this indexing would only happen with str and int args. IDK, maybe this should stay this way though: if you pass in an Expr, use that as the value. Otherwise, assume you are trying to index somehow into the existing table. But just wanted to bring it up to confirm this is expected.

### What version of ibis are you using?

8.0.0

### What backend(s) are you using, if any?

NA

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by jcrist** [comment]
Thanks for opening this - this is indeed a bug and was fixed in `the-epic-split` in #8106. We only have a test added for `.filter`. I'll push up a test to that branch to also cover `.select`.

**Comment by jcrist** [comment]
Closing as fixed in `the-epic-split` branch.
